pkg/apis/workflow/v1alpha1/item.go

```diff
@@ -58,15 +60,15 @@ func (i *Item) UnmarshalJSON(value []byte) error {
 }
 
 func (i *Item) String() string {
-	jsonBytes, err := json.Marshal(i)
+	x, err := json.Marshal(i) // this produces a normalised string, e.g. white-space
 	if err != nil {
 		panic(err)
 	}
 	// this convenience to remove quotes from strings will cause many problems
-	if jsonBytes[0] == '"' {
-		return string(jsonBytes[1 : len(jsonBytes)-1])
+	if x[0] == '"' {
+		return jsonutil.Fix(string(x[1 : len(x)-1]))
 	}
-	return string(jsonBytes)
+	return jsonutil.Fix(string(x))
```

Comment:
Suggestion:
Can it be the common wrapper function `json.Marshal` which contains the above definition? So, the developer will call the wrapper function instead of call `json.Marshal` and `jsonutil.fix`.